tiledb/sm/query/reader.cc

```diff
@@ -583,10 +583,10 @@ void Reader::compute_result_space_tiles(
       // Include this fragment in the space tile
       auto frag_domain = frag_tile_domains[f].domain_slice();
       auto frag_idx = frag_tile_domains[f].id();
-      result_space_tile.frag_domains_.emplace_back(frag_idx, frag_domain);
+      result_space_tile.append_frag_domain(frag_idx, frag_domain);
       auto tile_idx = frag_tile_domains[f].tile_pos(coords);
       ResultTile result_tile(frag_idx, tile_idx, domain);
-      result_space_tile.result_tiles_[frag_idx] = result_tile;
+      result_space_tile.set_result_tile(frag_idx, result_tile);
```

Comment:
Do we need to be concerned that this has a time complexity of logN because it is inserting into a sorted list? I don't know exactly how many result tiles we typically insert, so this may be negligble.

Additionally, if we are frequently inserting result_tile's with an increasing frag_idx, we could modify `set_result_tile` to use `result_tiles_.emplace_hint(--result_tiles_.end(), frag_idx, result_tile)`.

Reply:
This does not change the original logic and complexity (`result_tiles_ was a map before as well). We did not see any perf issues with that.